### PR TITLE
[DOCS-7980] Remove buffer in OP docs

### PR DESCRIPTION
**Pull request by maycmlee**
### What does this PR do? What is the motivation?

There isn't a buffer option for OP 2.0 so removing it from the docs.

[DOCS-7980](https://datadoghq.atlassian.net/browse/DOCS-7980)

### Merge instructions

- [ ] Please merge after reviewing

### Additional notes

[DOCS-7980]: https://datadoghq.atlassian.net/browse/DOCS-7980?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: content/en/observability_pipelines/best_practices_for_scaling_observability_pipelines.md [hunk 1]
```diff
@@ -51,7 +51,7 @@ Observability Pipelines Worker runs on modern CPU architectures. X86_64 architec
 
 ### Memory sizing
 
-Due to Observability Pipelines Worker's affine type system, memory is rarely constrained for Observability Pipelines Worker workloads. Therefore, Datadog recommends ≥2 GiB of memory per vCPU minimum. Memory usage increases with the number of sinks due to the in-memory buffering and batching. If you have a lot of sinks, consider increasing the memory or switching to disk buffers.
+Due to Observability Pipelines Worker's affine type system, memory is rarely constrained for Observability Pipelines Worker workloads. Therefore, Datadog recommends ≥2 GiB of memory per vCPU minimum. Memory usage increases with the number of sinks due to the in-memory buffering and batching. If you have a lot of sinks, consider increasing the memory.
 
 ### Disk sizing
```
Comment (maycmlee) [inline content/en/observability_pipelines/best_practices_for_scaling_observability_pipelines.md:56]: Do we want to remove this section as well?

Comment (maycmlee) [inline content/en/observability_pipelines/best_practices_for_scaling_observability_pipelines.md:56]: @ckelner do we want to remove this "Disk Sizing" section as well?

Comment (ckelner) [reply content/en/observability_pipelines/best_practices_for_scaling_observability_pipelines.md:56]: Per https://vector.dev/docs/setup/going-to-prod/sizing/#disks it states:
> Sizing disks is only relevant if you’re using Vector’s disk buffers.

But I feel like we should have some kind of guidance here, even w/o disk buffers.
Let me ask the OP backroom, I have a thread on this topic.

File: content/en/observability_pipelines/legacy/production_deployment_overview.md [hunk 2]
```diff
@@ -89,29 +88,6 @@ Data processing is shifted off your nodes and onto remote aggregator nodes. Remo
 
 See [Aggregator Architecture][5] for more details.
 
-## Buffering data
```
Comment (ckelner) [inline content/en/observability_pipelines/legacy/production_deployment_overview.md:92]: I think removing this for now is good, but we should write up something about the "built in default" buffering (and remove language about being able to configure it). 

We can probably pull something together from here: https://vector.dev/docs/about/under-the-hood/architecture/buffering-model/ then likely have eng review it. 

I can draft something in a google doc if you like

**Review by ckelner — COMMENTED** [review]
Just commenting for now, lets wait on OP Eng to comment on this: https://github.com/DataDog/documentation/pull/23270/files#r1607013668

File: content/en/observability_pipelines/legacy/production_deployment_overview.md (hunk 2 above)
Comment (maycmlee) [reply content/en/observability_pipelines/legacy/production_deployment_overview.md:92]: Sure, if you can draft something that'd be great. Thanks!

Comment (ckelner) [reply content/en/observability_pipelines/legacy/production_deployment_overview.md:92]: Check it out here May: https://docs.google.com/document/d/1UKspJYeXbvY5tFdlJUJgjvJ2ngLfg36MAkKVz6YCQvo/edit

Want to post in the #obs-pipelines-backroom channel for a look?

Comment (ckelner) [reply content/en/observability_pipelines/legacy/production_deployment_overview.md:92]: Oh jeez! These are the "legacy" 1.0 docs. We can disregard. Leave this as-is.

File: content/en/observability_pipelines/best_practices_for_scaling_observability_pipelines.md (hunk 1 above)
Comment (ckelner) [reply content/en/observability_pipelines/best_practices_for_scaling_observability_pipelines.md:56]: @maycmlee I think for "Disk Sizing" we just say "30GB minimum" and leave it at that.

**Review by ckelner — CHANGES_REQUESTED** [review]
Let's update disk buffering, and let's revert the delete for the legacy docs.

File: content/en/observability_pipelines/legacy/production_deployment_overview.md (hunk 2 above)
Comment (maycmlee) [reply content/en/observability_pipelines/legacy/production_deployment_overview.md]: Sorry that was my fault, I thought I had stuck to the current docs. I reverted the change.

File: content/en/observability_pipelines/best_practices_for_scaling_observability_pipelines.md [hunk 3]
```diff
@@ -51,7 +51,7 @@ Observability Pipelines Worker runs on modern CPU architectures. X86_64 architec
 
 ### Memory sizing
 
-Due to Observability Pipelines Worker's affine type system, memory is rarely constrained for Observability Pipelines Worker workloads. Therefore, Datadog recommends ≥2 GiB of memory per vCPU minimum. Memory usage increases with the number of sinks due to the in-memory buffering and batching. If you have a lot of sinks, consider increasing the memory or switching to disk buffers.
+Due to Observability Pipelines Worker's affine type system, memory is rarely constrained for Observability Pipelines Worker workloads. Therefore, Datadog recommends ≥2 GiB of memory per vCPU minimum. Memory usage increases with the number of sinks due to the in-memory buffering and batching. If you have a lot of sinks, consider increasing the memory.
```
Comment (ckelner) [inline content/en/observability_pipelines/best_practices_for_scaling_observability_pipelines.md:54]:
```suggestion
Due to Observability Pipelines Worker's affine type system, memory is rarely constrained for Observability Pipelines Worker workloads. Therefore, Datadog recommends ≥2 GiB of memory per vCPU minimum. Memory usage increases with the number of destinations due to the in-memory buffering and batching. If you have a lot of destinations, consider increasing the memory.
```

We call them destinations now right?

**Review by ckelner — CHANGES_REQUESTED** [review]
just thought of something, we don't call them sinks anymore right, should we change to destinations?

File: content/en/observability_pipelines/best_practices_for_scaling_observability_pipelines.md (hunk 3 above)
Comment (maycmlee) [reply content/en/observability_pipelines/best_practices_for_scaling_observability_pipelines.md]: Yes, good catch

File: content/en/observability_pipelines/best_practices_for_scaling_observability_pipelines.md (hunk 1 above)
Comment (maycmlee) [reply content/en/observability_pipelines/best_practices_for_scaling_observability_pipelines.md:56]: @ckelner could you let me know if the update below is what you are suggesting? I'm assuming the example caclulation is still correct?

> For example, an 8 vCPU machine would require 288 GiB of disk space (10 MiB * 60 seconds * 60 minutes * 8 vCPUs).

Comment (maycmlee) [reply content/en/observability_pipelines/best_practices_for_scaling_observability_pipelines.md:56]: Chatted on zoom. Remove the entire section and just add the minimum required sentence.

**Review by ckelner — APPROVED** [review]

**Review by hestonhoffman — APPROVED** [review]

**Pull request merged**